### PR TITLE
- Update PSK text with warning about 1.2/1.3

### PR DESCRIPTION
**Pull request by ekr**
- Add Bleichenbacher reference

### CODE REVIEW

**Review by chris-wood — APPROVED** [review]
s/although they both based on HMAC/although they are both based on HMAC

Looks good otherwise.

File: rfc8446.xml [hunk 1]
```diff
@@ -7393,32 +7403,29 @@ this form of attack by treating the cases where there is no
...
+TLS 1.3 takes a conservative approach to PSKs by binding them to a
+specific KDF. By contrast, TLS 1.2 allows PSKs to be used with any
+hash function and the TLS 1.2 PRF. Thus, any PSK which is used with
+both TLS 1.2 and TLS 1.3 must be used with only one hash in TLS 1.3,
+which is less than optimal if users want to provision a single PSK.
+The constructions in TLS 1.2 and TLS 1.3 are different, although they
+are both based on HMAC. While there is no known way in which the same
+PSK might produce related output in both versions, only limited
+analysis has been done. Implementations can ensure safety from
+cross-protocol key collisions by not reusing PSKs between TLS 1.3 and
```
Comment (kaduk) [inline rfc8446.xml:7419]: Is key collisions the only form of attack we're worried about, or is there a different class of attack that involves simultaneously analyzing ciphertexts produced using the same input key for different KDFs?

File: rfc8446.xml [hunk 2]
```diff
@@ -7393,32 +7403,29 @@ this form of attack by treating the cases where there is no
...
+TLS 1.2.
```
Comment (kaduk) [inline rfc8446.xml:7420]: I might consider adding something along the lines of "Developing a robust and fully analyzed scheme for allowing existing deployed TLS 1.2 PSKs to be safely used with multiple hashes in TLS 1.3 remains a topic for futrue work."
But it's far from clear that it's needed.

File: rfc8446.xml (hunk 1 above)
Comment (chris-wood) [reply rfc8446.xml:7419]: I’m not aware of any such attack. @ekr?

File: rfc8446.xml (hunk 2 above)
Comment (chris-wood) [reply rfc8446.xml:7420]: IMO omission of this text sends the same signal as its inclusion. So I’d err on the side of less text.

File: rfc8446.xml (hunk 1 above)
Comment (ekr) [reply rfc8446.xml:7419]: I am not, but why not just say 
"cross-protocol related output"

File: rfc8446.xml (hunk 2 above)
Comment (ekr) [reply rfc8446.xml:7420]: I am fine either way, but the chairs told me not too.

Comment (kaduk) [reply rfc8446.xml:7420]: I am fine leaving it out.

File: rfc8446.xml (hunk 1 above)
Comment (kaduk) [reply rfc8446.xml:7419]: That works for me.

Comment (chris-wood) [reply rfc8446.xml:7419]: 👍

**Pull request closed without merging**